pkg/util/apiserver/cors.go

```diff
@@ -3,19 +3,43 @@
 
 package apiserver
 
-import "net/http"
+import (
+	"net/http"
+	"strings"
+
+	"github.com/verrazzano/verrazzano-operator/pkg/util"
+
+	"github.com/verrazzano/verrazzano-operator/pkg/api/instance"
+)
 
 // EnableCors adds headers necessary for the browser running the UI to be able to call the API server
-func EnableCors(w *http.ResponseWriter) {
-	(*w).Header().Set("Access-Control-Allow-Origin", "*")
+func EnableCors(req *http.Request, w *http.ResponseWriter) {
+	origin := req.Header.Get("Origin")
+	(*w).Header().Set("Access-Control-Allow-Origin", getAllowedOrigin(origin))
 	// The web UI expects to be able to see X-Total-Count, which is uses for paging result sets
 	(*w).Header().Set("Access-Control-Expose-Headers", "X-Total-Count")
 }
 
 // SetupOptionsResponse creates the headers that are needed for a HTTP OPTIONS request
 // The web UI makes an OPTIONS request before each GET/POST/etc.
 func SetupOptionsResponse(w *http.ResponseWriter, req *http.Request) {
-	(*w).Header().Set("Access-Control-Allow-Origin", "*")
+	origin := req.Header.Get("Origin")
+	(*w).Header().Set("Access-Control-Allow-Origin", getAllowedOrigin(origin))
 	(*w).Header().Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS, PUT, DELETE")
 	(*w).Header().Set("Access-Control-Allow-Headers", "Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization")
 }
+
+func getAllowedOrigin(origin string) string {
```

Comment:
isn't this going to be called every time? seems like it would be better to calculate it once and store it?

Reply:
this logic was more complicated by the fact that I was allowing multiple override values to be supplied, which we really don't need. I changed it to allow only a single value, and that made the logic much simpler, so we don't have to worry about it being called multiple times